### PR TITLE
Feat: app startup hooks

### PR DESCRIPTION
**Pull request by inckie**
Application startup hooks re-introduced.
Both executeOnStartup and executeOnApplicationReady are supported.

### Checklist
- [x] I've provided a readable title for the PR
- [x] I've provided a detailed description
- [x] The PR is scoped to a single task/feature
- [ ] I've included relevant tests (if tests are not included please provide the reason why they aren't)

- [x] My PR is not introducing a UI change
- [ ] My PR is introducing UI changes and I provided all screenshots in the PR description

- [x] My PR is a part of a new feature or a feature improvement
    - [ ] I've provided a link in the description to the relevant card in the Zapp cycle feature rollout Trello board.
- [ ] My PR is a bug fix
    - [ ] I provided a link in the description to the relevant Jira ticket  or provided the following in the PR description:
        - steps to reproduce a bug
        - expected result

### CODE REVIEW

File: android_quickbrick_app/src/main/java/com/applicaster/ui/application/BaseApplication.java [hunk 1]
```diff
@@ -4,7 +4,6 @@
 import android.util.Log;
 
 import com.applicaster.app.APProperties;
-import com.applicaster.app.CustomApplication;
```
Comment (ranmeirman) [inline android_quickbrick_app/src/main/java/com/applicaster/ui/application/BaseApplication.java:7]: why is this needed?

File: android_quickbrick_app/src/main/java/com/applicaster/ui/activities/MainActivity.java [hunk 2]
```diff
@@ -218,7 +227,31 @@ private synchronized void preloadStepComplete(PreloadStep step) {
         preloadStateManager.setStepComplete(step);
 
         if (preloadStateManager.isPreloadComplete()) {
-            runOnUiThread(this::initializeUILayer);
+            runOnUiThread(this::executeOnApplicationReadyHooks);
+        }
+    }
+
+    private void executeOnStartupHooks(@NonNull Runnable continuation) {
+        List<ApplicationLoaderHookUpI> hookPluginList = PluginManager.getInstance().getHookPluginList();
```
Comment (ranmeirman) [inline android_quickbrick_app/src/main/java/com/applicaster/ui/activities/MainActivity.java:235]: I am missing the higher context here, do we always have plugin config here?

File: android_quickbrick_app/src/main/java/com/applicaster/ui/application/BaseApplication.java (hunk 1 above)
Comment (inckie) [reply android_quickbrick_app/src/main/java/com/applicaster/ui/application/BaseApplication.java:7]: its was an unused import

File: android_quickbrick_app/src/main/java/com/applicaster/ui/activities/MainActivity.java (hunk 2 above)
Comment (inckie) [reply android_quickbrick_app/src/main/java/com/applicaster/ui/activities/MainActivity.java:235]: we do have one that was initial/loaded before. It that callback happens before data been loaded

Comment (inckie) [reply android_quickbrick_app/src/main/java/com/applicaster/ui/activities/MainActivity.java:235]: https://github.com/applicaster/applicaster-android-sdk/blob/dev/src/main/java/com/applicaster/plugin_manager/hook/ApplicationLoaderHookUpI.java
